runners/apex/src/main/java/org/apache/beam/runners/apex/translation/operators/ApexParDoOperator.java

```diff
@@ -255,6 +252,18 @@ public void process(ApexStreamTuple<WindowedValue<Iterable<?>>> t) {
   private Iterable<WindowedValue<InputT>> processElementInReadyWindows(WindowedValue<InputT> elem) {
     try {
       pushbackDoFnRunner.startBundle();
+      if (currentKeyStateInternals != null) {
+        InputT value = elem.getValue();
+        // TODO: generic key extraction
```

Comment:
What is this referring to? I don't think there are other cases in Beam for extracting a key, except for KeyedWorkItem and KV.